### PR TITLE
Make some of representation tests single-threaded

### PR DESCRIPTION
**Pull request by eupp**
Several improvements of representation tests:
1. Make more representation tests to inherit from `BaseTraceRepresentationTest` (formerly `BaseFailingTest`), unify test set-up, simplify tests. 
2. Make all of the tests inheriting from `BaseTraceRepresentationTest` single-threaded, thus factoring out possible spurious test failures due to changes in thread switching order in the generated traces.

### CODE REVIEW

File: src/jvm/test/resources/expected_logs/var_handle/varhandle_int_representation.txt [hunk 1]
```diff
@@ -1,56 +1,32 @@
...
-| ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| -------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+|                                                                            Thread 1                                                                            |
+| -------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| operation(): 1                                                                                                                                                 |
+|   actionsForTrace() at BaseTraceRepresentationTest.operation(BaseTraceRepresentationTest.kt:27)                                                                |
```
[the remark this answers is not in the document]
Comment (eupp) [reply src/jvm/test/resources/expected_logs/var_handle/varhandle_int_representation.txt:20]: Yes, it is an `abstract fun` in `BaseTraceRepresentationTest`. 
All other tests inheriting from `BaseTraceRepresentationTest` need only to override this `actionsForTrace` method.

File: src/jvm/test/resources/expected_logs/var_handle/varhandle_char_representation.txt [hunk 2]
```diff
@@ -1,56 +1,32 @@
...
+|     number.compareAndSet(1,1): true at VarHandleCharRepresentationTest.actionsForTrace(VarHandleRepresentationTests.kt:179)                                       |
+|     number.set(2) at VarHandleCharRepresentationTest.actionsForTrace(VarHandleRepresentationTests.kt:180)                                                         |
+|     VarHandleCharRepresentationTest.staticNumber.READ: 3 at VarHandleCharRepresentationTest.actionsForTrace(VarHandleRepresentationTests.kt:182)                  |
+|     VarHandleCharRepresentationTest.staticNumber.compareAndSet(3,1): true at VarHandleCharRepresentationTest.actionsForTrace(VarHandleRepresentationTests.kt:182) |
+|     VarHandleCharRepresentationTest.staticNumber.set(3) at VarHandleCharRepresentationTest.actionsForTrace(VarHandleRepresentationTests.kt:183)                   |
+|     array.READ: CharArray#1 at VarHandleCharRepresentationTest.actionsForTrace(VarHandleRepresentationTests.kt:185)                                               |
+|     CharArray#1[1].compareAndSet(3,1): false at VarHandleCharRepresentationTest.actionsForTrace(VarHandleRepresentationTests.kt:185)                              |
```
[the remark this answers is not in the document]
Comment (eupp) [reply src/jvm/test/resources/expected_logs/var_handle/varhandle_char_representation.txt]: Fixed, please have a look.

File: src/jvm/test/resources/expected_logs/var_handle/varhandle_int_representation.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (ndkoval) [reply src/jvm/test/resources/expected_logs/var_handle/varhandle_int_representation.txt:20]: Can't you override the `operation` function itself?

[the remark this answers is not in the document]
Comment (eupp) [reply src/jvm/test/resources/expected_logs/var_handle/varhandle_int_representation.txt:20]: Agreed to override `operation` and use always failing verifier.

**Review by ndkoval — APPROVED** [review]

**Comment by ndkoval** [comment]
Approved 🥳 

@eupp please merge when the tests pass

**Pull request merged**